File: author_page_generate.py
import json
import os

def generate_pages():
    try:
        # Load data from JSON file
        with open('data/publications.json', 'r') as f:
            data = json.load(f)
    except FileNotFoundError:
        print("The file data/publications.json was not found.")
        return

    except json.JSONDecodeError:
        print("Error decoding JSON from the file.")
        return

    # Create the output directory if it doesn't exist
    if not os.path.exists('output'):
        os.makedirs('output')

    # Iterate through each author ID in the JSON data
    for authorid, author_publications in data.items():
        # Generate HTML content
        html_content = f"""
        <!DOCTYPE html>
        <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1.0">
            <title>Author Details - {authorid}</title>
        </head>
        <body>
            <h1>Author Details - {authorid}</h1>
        """
        
        for publication in author_publications:
            html_content += f"""
            <ul>
                <li><strong>Title:</strong> {publication['Title']}</li>
                <li><strong>Author:</strong> {publication['Author']}</li>
                <li><strong>Descriptor:</strong> {publication.get('Descriptor', 'N/A')}</li>
                <li><strong>Genre:</strong> {publication['Genre']}</li>
                <li><strong>Language:</strong> {publication['Language']}</li>
                <li><strong>Pub ID:</strong> {publication['Pub_id']}</li>
                <li><strong>Pubdate:</strong> {publication['Pubdate']}</li>
                <li><strong>Publisher:</strong> {publication['Publisher']}</li>
                <li><strong>Translation:</strong> {publication['Translation']}</li>
            </ul>
            <hr>
            """
        
        html_content += """
        </body>
        </html>
        """
        
        # Write HTML content to a file
        output_file_path = os.path.join('output', f'{authorid}.html')
        with open(output_file_path, 'w') as output_file:
            output_file.write(html_content)
        
        print(f"Output page generated for Author ID: {authorid}")

# Generate pages for all authors
generate_pages()

File: author_page_generate.js
const fs = require('fs');
const path = require('path');

function generatePages() {
    let data;
    
    // Load data from JSON file
    try {
        const dataPath = path.join('data', 'publications.json');
        const jsonData = fs.readFileSync(dataPath, 'utf-8');
        data = JSON.parse(jsonData);
    } catch (err) {
        if (err.code === 'ENOENT') {
            console.error("The file data/publications.json was not found.");
        } else if (err instanceof SyntaxError) {
            console.error("Error decoding JSON from the file.");
        } else {
            console.error("An unknown error occurred:", err);
        }
        return;
    }

    // Create the output directory if it doesn't exist
    const outputDir = path.join(__dirname, 'output');
    if (!fs.existsSync(outputDir)) {
        fs.mkdirSync(outputDir);
    }

    // Iterate through each author ID in the JSON data
    for (const authorid in data) {
        if (data.hasOwnProperty(authorid)) {
            const authorPublications = data[authorid];
            
            // Generate HTML content
            let htmlContent = `
            ---
            layout: defaultau
            title: ${authorid} Author Name 
            permalink: /${authorid}
            ---
            <html lang="en">
            <head>
                <meta charset="UTF-8">
                <meta name="viewport" content="width=device-width, initial-scale=1.0">
                <title>Author Details - ${authorid}</title>
            </head>
            <body>
                <h1>Author Details - ${authorid}</h1>
            `;
            
            for (const publication of authorPublications) {
                htmlContent += `
                <ul>
                    <li><strong>Title:</strong> ${publication['Title']}</li>
                    <li><strong>Genre:</strong> ${publication['Genre']}</li>
                    <li><strong>Language:</strong> ${publication['Language']}</li>
                    <li><strong>Publisher:</strong> ${publication['Publisher']}</li>
                    <li><strong>Publication City:</strong> ${publication['Pub_id']}</li>
                    <li><strong>Publication Date:</strong> ${publication['Pubdate']}</li>
                    <li><strong>Translation:</strong> ${publication['Translation']}</li>
                </ul>
                <hr>
                `;
            }
            
            htmlContent += `
            </body>
            </html>
            `;
            
            // Write HTML content to a file
            const outputFilePath = path.join(outputDir, `${authorid}.md`);
            fs.writeFileSync(outputFilePath, htmlContent);

            console.log(`Output page generated for Author ID: ${authorid}`);
        }
    }
}

// Generate pages for all authors
generatePages();
